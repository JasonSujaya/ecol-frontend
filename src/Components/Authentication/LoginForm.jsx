import React from "react";

class LoginForm extends React.Component {
  render() {
    return <div>Login stuff here</div>;
  }
}

export default LoginForm;
