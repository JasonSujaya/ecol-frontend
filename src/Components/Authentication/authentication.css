.formContainer {
    display: flex;
    flex-wrap: wrap;
    flex-direction: column;
}

.authenticationContainer{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin-top: 20px;
}