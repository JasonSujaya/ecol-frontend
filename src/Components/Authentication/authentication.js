import axios from "axios";

export const storeToken = (token) => {
  localStorage.token = token;
};

export const checkForToken = () => {
  let tokenValue = localStorage.token;
  if (tokenValue != null) {
    if (tokenValue.length != 0) {
      return true;
    } else {
      return false;
    }
  } else {
    return false;
  }
};

export const login = async (_email, _password) => {
  const user = {
    username: _email,
    password: _password,
  };

  const response = await axios.post(
    `http://127.0.0.1:8000/api/user/login/`,
    user,
    {
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  return response;
};

export const formsNotEmpty = (_email, _password) => {
  if (_email.length != 0 && _password.length != 0) {
    return true;
  } else {
    return false;
  }
};

export const signUp = async (_email, _first_name, _last_name, _password) => {
  const user = {
    email: _email,
    first_name: _first_name,
    last_name: _last_name,
    password: _password,
  };

  const response = await axios.post(
    `http://127.0.0.1:8000/api/user/profiles/`,
    user,
    {
      headers: {
        "Content-Type": "application/json",
      },
    }
  );

  return response;
};

export const inputNotEmpty = (_input) => {
  if (_input.length != 0) {
    return true;
  } else {
    return false;
  }
};

export const checkPassword = (password) => {
  let re = "";
  if (password != "") {
    if (password.length < 6) {
      return "Error: Password must contain at least six characters!";
    }
    re = /[0-9]/;
    if (!re.test(password)) {
      return "Error: password must contain at least one number (0-9)!";
    }
    re = /[a-z]/;
    if (!re.test(password)) {
      return "Error: password must contain at least one lowercase letter (a-z)!";
    }
    re = /[A-Z]/;
    if (!re.test(password)) {
      return "Error: password must contain at least one uppercase letter (A-Z)!";
    }
  } else {
    return "Error: Please check that you've entered and confirmed your password!";
  }

  return true;
};
