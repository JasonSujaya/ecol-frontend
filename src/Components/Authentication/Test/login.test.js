import axios from "axios";
import { shallow, mount } from "enzyme";

import LoginForm from "../LoginForm";
import { login, formsNotEmpty } from "../authentication.js";

function resolvePromiseOnSimulate() {
  return new Promise((resolve) => {
    setTimeout(resolve, 0);
  });
}

const setUp = () => {
  axios.post.mockResolvedValue({
    data: {
      token: "someMockTokenHere",
    },
  });
};

describe("Login function", () => {
  it("Returns JWS token when the sign in is successful", async () => {
    // Arrange
    setUp();

    // Act
    const response = await login("mock@email.com", "mockpassword");

    // Assert
    expect(response.data.token).toEqual("someMockTokenHere");
  });

  it("Expect function to throw error when email and/or password is inputted", () => {
    // Arrange
    setUp();

    // Act & Assert
    expect(() => {
      formsNotEmpty("", "");
    }).toThrow("Email and/or password is emptyy");

    expect(() => {
      formsNotEmpty("Hello@gmail.com", "");
    }).toThrow("Email and/or password is emptyy");

    expect(() => {
      formsNotEmpty("", "Password");
    }).toThrow("Email and/or password is emptyy");
  });
});
