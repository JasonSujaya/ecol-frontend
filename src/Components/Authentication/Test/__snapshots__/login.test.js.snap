// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign In component Sign in renders correctly 1`] = `ShallowWrapper {}`;
